src/test_l1_dsAudio.c

```diff
@@ -9954,6 +9954,117 @@ void test_l1_dsAudio_negative_dsGetHDMIARCPortId(void) {
+			DS_ASSERT_AUTO_TERM_NUMERICAL(dsSetAudioMixerLevels(handle[i], audioInput, volume), dsERR_NONE);
+		}
+	}
+
+	// Step 04: Terminate the Audio Port using dsAudioPortTerm()
+	UT_ASSERT_EQUAL(dsAudioPortTerm(), dsERR_NONE);
+
+	// End of the test
+	UT_LOG("\n Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief Ensure dsSetAudioMixerLevels() returns correct error codes during negative scenarios.
+*
+* **Test Group ID:** Basic: 01@n
+* **Test Case ID:** 164@n
+*
+* **Dependencies:** None@n
+* **User Interaction:** None
+*
+* **Test Procedure:**@n
+* |Variation / Step|Description|Test Data|Expected Result|Notes|
+* |:--:|-----------|----------|--------------|-----|
+* |01|Call dsSetAudioMixerLevels() without prior initialization of Audio Ports | handle = [inValid handle], aInput = dsAudioInputPrimary, volume= 75 | dsERR_NOT_INITIALIZED | Should Pass |
+* |02|Initialize Audio Ports using dsAudioPortInit() | | dsERR_NONE | Should Pass |
+* |03|Call dsSetAudioMixerLevels() with an invalid handle but with valid audioInput and volume level | handle = [inValid handle], aInput = dsAudioInputPrimary, volume = 75 | dsERR_INVALID_PARAM | Should Pass |
+* |04|Get the audio port handle for valid audio port type and valid index | type, index = [Loop through kPorts] , handle = [valid handle] | dsERR_NONE | Valid port handle must be returned |
+* |05|Call dsSetAudioMixerLevels() by looping through acquired handles with invalid audioInput and valid volumelevels | handle = [valid handle], aInput = dsAudioInputMax , volume: 75 | dsERR_INVALID_PARAM | Should Pass |
+* |06|Call dsSetAudioMixerLevels() by looping through acquired handles with valid audioInput and invalid volumelevels | handle = [valid handle], aInput = dsAudioInputPrimary , volume: 105 | dsERR_INVALID_PARAM | Should Pass |
+* |07|Terminate the Audio Port using dsAudioPortTerm() | | dsERR_NONE | Clean up after test |
+* |08|Call dsSetAudioMixerLevels() after termination | handle= [valid handle], aInput= dsAudioInputMax, volume= 75 | dsERR_NOT_INITIALIZED | Should Pass |
+*
+* @note Testing for the `dsERR_OPERATION_NOT_SUPPORTED` and `dsERR_OPERATION_FAILED` might be challenging since it requires a specific scenario where the attempted operation is not supported or has failed respectively.
+*
+*/
+void test_l1_dsAudio_negative_dsSetAudioMixerLevels(void) {
+	gTestID = 164;
+	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+	int result, invalid_vol_level = 105;;
+	intptr_t  handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
+
+	// Step 01: Call dsSetAudioMixerLevels() without prior initialization of Audio Port
+	UT_ASSERT_EQUAL(dsSetAudioMixerLevels(-1, dsAudioInputPrimary, 75), dsERR_NOT_INITIALIZED);
+
+	// Step 02: Initialize Audio Port using dsAudioPortInit()
+	UT_ASSERT_EQUAL(dsAudioPortInit(), dsERR_NONE);
+
+	// Step 03: Call dsSetAudioMixerLevels() with an invalid port handle
+	DS_ASSERT_AUTO_TERM_NUMERICAL(dsSetAudioMixerLevels(handle[0], dsAudioInputPrimary, 75), dsERR_INVALID_PARAM);
```

Comment:
Use UT Macros correctly please , Terminate should be at the end of your function not during.


Remove all uses of ->

DS_ASSERT_AUTO_TERM_NUMERICAL -> all tests need to fall through. -> UT_ASSERT
DS_ASSERT_AUTO_TERM_NOT_EQUAL -> all tests need to fall through -> UT_ASSERT_NOT_EQUAL
DS_ASSERT_AUTO_TERM_STRING -> remove and fall through.  -> UT_ASSERT_STRING

Reply:
Addressed